File: 240123/문자에 따른 명령 2/text-based-commands2.py
dir = list(input())
x, y = 0, 0
dir_num = 3
dx = [1, 0, -1, 0]
dy = [0, -1, 0, 1]
for i in range(len(dir)):
    if dir[i] == 'R':
        dir_num = (dir_num + 1) % 4
    elif dir[i] == 'L':
        dir_num = (dir_num + 3) % 4
    else:
        x += dx[dir_num]
        y += dy[dir_num]
print(x, y)

File: 240909/문자에 따른 명령 2/text-based-commands2.js
const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("");

let dx = [1, 0, -1, 0]; let dy = [0, -1, 0, 1];
let dirNum = 3;

let [x, y] = [0, 0];

for (let i = 0; i < input.length; i++) {
    let dir = input[i];
    if (dir === "L") {
        dirNum = (dirNum + 3) % 4;
    } else if (dir === "R") {
        dirNum = (dirNum + 1) % 4;
    } else {
        x += dx[dirNum]; y += dy[dirNum];
    }
}

console.log(x, y);